rapyuta_io_sdk_v2/utils.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+# Copyright 2024 Rapyuta Robotics
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# from rapyuta_io_sdk_v2.config import Configuration
+import http
+import json
+from typing import Any, Dict
+
+import httpx
+from munch import Munch
+
+from rapyuta_io_sdk_v2.exceptions import HttpAlreadyExistsError, HttpNotFoundError
+
+
+def munchify_projects(response: httpx.Response) -> Munch:
+    data = response.json()
+    projects = []
+    for item in data.get("items", []):
+        project = item.get("metadata", {}).get("name")
+        project_guid = item.get("metadata", {}).get("projectGUID")
+        if project and project_guid:
+            projects.append({"project": project, "project_guid": project_guid})
+    return Munch({"projects": projects})
+
+
+def validate_auth_token(config: Any) -> Dict:
+    try:
+        client = config.sync_client()
+        user = client.get_authenticated_user()
+        return user
+    except Exception:
+        raise
+
+
+def handle_server_errors(response: httpx.Response):
+    status_code = response.status_code
+
+    if status_code < 400:
+        return
+
+    err = ""
+    try:
+        err = response.json().get("error")
+    except json.JSONDecodeError:
+        err = response.text
+
+    # 404 Not Found
+    if status_code == http.HTTPStatus.NOT_FOUND:
+        raise HttpNotFoundError(err)
+    # 405 Method Not Allowed
+    if status_code == http.HTTPStatus.METHOD_NOT_ALLOWED:
+        raise Exception("method not allowed")
```

Comment:
suggestion: Consider implementing the Exception classes for other error codes as well. It will help with better exception handling in the clients of the SDK.